### PR TITLE
Allow to use function as liveSearchStyle

### LINKED ISSUE #2042: Support custom filtering
*@yoeriboven commented on May 7, 2018, 10:40 AM UTC:*

It might be a good idea to let the user do the filtering. Currently I'm working on a project in which I want the search to be done in the _value_ of the option or in the text with _startsWith_ as the option.

This could be done like this:

```
$('.selectpicker').on('typed.bs.select', function (input) {
  // Filter results here

  return results; // Array of results
});

```

Any suggestions?

*This issue was moved by [caseyjhol](https://github.com/caseyjhol) from [snapappointments/bootstrap-select-temp#56](https://github.com/snapappointments/bootstrap-select-temp/issues/56).*

### PR DESCRIPTION
**Pull request by cedk**
Fix #2042

### CODE REVIEW

**Comment by cedk** [comment]
ping

**Comment by cedk** [comment]
Hi @caseyjhol, do you think there are chances that this proposal would be merged?
I think it provide a powerful API to customize the behavior of the search.
Thanks.

File: js/bootstrap-select.js [hunk 1]
```diff
@@ -418,7 +418,9 @@
         if (normalize) string = normalizeToBase(string);
         string = string.toUpperCase();
 
-        if (method === 'contains') {
+        if (method instanceof Function) {
```
Comment (caseyjhol) [inline js/bootstrap-select.js]: Please change this to `typeof method === 'function'` instead.

**Review by caseyjhol — COMMENTED** [review]
@cedk Thanks for this! If you make the requested changes I can get this merged into the next minor release.

**Comment by cedk** [comment]
Could this be merged?

**Comment by caseyjhol** [comment]
Thanks for this PR - sorry it took so long to get merged.

**Pull request merged**